src/mcts/node.cc

```diff
@@ -230,6 +230,17 @@ void Node::MakeTerminal(GameResult result) {
   }
 }
 
+void Node::MakeNotTerminal() {
+  is_terminal_ = false;
```

Comment:
I think we should re-calculate Q and D here as well. 
Also are we confident that n_in_flight/best_cached_child/visited_policy are all correct?

Reply:
Added Q and D recomputation in https://github.com/LeelaChessZero/lc0/pull/822/commits/802e1ff48faec449d3e93af17e6caa5945ac2f91

The other values are updated in the usual FinalizeScoreUpdate calls from backup.

Reply:
Seems fine.